providers/gce/gce_instances.go

```diff
@@ -551,19 +551,29 @@ func (g *Cloud) AliasRangesByProviderID(providerID string) (cidrs []string, err
 		for _, r := range networkInterface.AliasIpRanges {
 			cidrs = append(cidrs, r.IpCidrRange)
 		}
-		ipv6Addr := getIPV6AddressFromInterface(networkInterface)
-		if ipv6Addr != "" {
-			// The podCIDR range is the first /112 subrange from the /96 assigned to
-			// the node
-			ipv6PodCIDR := fmt.Sprintf("%s/112", ipv6Addr)
-			cidrs = append(cidrs, ipv6PodCIDR)
-		} else {
-			klog.Infof("No IPv6 addresses found for %s", providerID)
+		ipv6Addr := g.GetIPV6Address(networkInterface)
+		if ipv6Addr != nil {
+			cidrs = append(cidrs, ipv6Addr.String())
 		}
 	}
 	return
 }
 
+// GetIPV6Address fetches the IPv6 addressses associated with a network interface.
+func (g *Cloud) GetIPV6Address(networkInterface *compute.NetworkInterface) *net.IPNet {
```

Comment:
this doesn't need to be a method, does it?

Reply:
`getIPV6AddressFromInterface` -> This is the method that is internally called from here and that has many usages across the codebase and is not exported. We could ideally export it and change all it's occurrences, but I didn't want to do that because it doesn't fall within the scope of this PR.